features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -44,19 +47,20 @@ internal class TelemetryEventHandler(
     private var totalEventsSeenInCurrentSession = 0
 
     @AnyThread
+    @Suppress("LongMethod")
     fun handleEvent(
-        event: RumRawEvent.SendTelemetry,
+        wrappedEvent: RumRawEvent.TelemetryEventWrapper,
         writer: DataWriter<Any>
     ) {
+        val event = wrappedEvent.event
         if (!canWrite(event)) return
 
         eventIDsSeenInCurrentSession.add(event.identity)
         totalEventsSeenInCurrentSession++
-
         sdkCore.getFeature(Feature.RUM_FEATURE_NAME)?.withWriteContext { datadogContext, eventBatchWriter ->
-            val timestamp = event.eventTime.timestamp + datadogContext.time.serverTimeOffsetMs
-            val telemetryEvent: Any? = when (event.type) {
-                TelemetryType.DEBUG -> {
+            val timestamp = wrappedEvent.eventTime.timestamp + datadogContext.time.serverTimeOffsetMs
+            val telemetryEvent: Any? = when (event) {
```

Comment:
does it become exhaustive `when`? The reason why I'm asking it because there are 2 sealed classes involved here, not only one. Will it be a compilation error if we add say `InternalTelemetry.Log.XXX` into classes definition?

Reply:
I think the way it behaves is that whenever you add a new type of the sealed class it will warn you that is not handled but will double check it.

Reply:
yes, but you now have 2 sealed base classes in this hierarchy, not one, that is why I'm curious.

Reply:
Yeah, I confirm whenever you add a new type to that sealed class it is only to crash at compile time saying that you need to be exhaustive.